src/app/parcels/ActionBar/Actions.tsx

```diff
@@ -98,19 +107,16 @@ const Actions: React.FC<Props> = ({
 }) => {
     const [selectedParcels, setSelectedParcels] = useState<ParcelsTableRow[]>([]);
     const [modalToDisplay, setModalToDisplay] = useState<ActionName | null>(null);
-    const [availableActionsForUserRole, setAvailableActionsForUserRole] = useState<ActionTypes[]>(
-        []
-    );
     const { role } = useContext(RoleUpdateContext);
 
-    useEffect(() => {
-        if (role === "volunteer") {
-            const actionList = availableActions.filter((action) => action.availableToVolunteers);
-            setAvailableActionsForUserRole(actionList);
-        } else {
-            setAvailableActionsForUserRole(availableActions);
+    const getAvailableActionsForUserRole = (): ActionTypes[] => {
+        if (role === null) {
+            return [];
         }
-    }, [role]);
+        return availableActions.filter((action) => {
+            return action.availableToRole.includes(role);
+        });
+    };
```

Comment:
i feel like this function doesn't need to be a function but a constant - then we won't have to call it twice?

Reply:
Good point - I've created a constant using this function because I need to exclude role = null because action.availableToRole does not include null (typing issue).